### PR TITLE
[SPARK-32190][PYTHON][DOCS] Development - Contribution Guide in PySpark

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to document PySpark specific contribution guides at "Development" section.

Here is the demo for reviewing quicker: https://hyukjin-spark.readthedocs.io/en/stable/development/contributing.html

### Why are the changes needed?

To have a single place for PySpark users, and better documentation.

### Does this PR introduce _any_ user-facing change?

Yes, it is a new documentation. See the demo linked above.

### How was this patch tested?

```bash
cd docs
SKIP_SCALADOC=1 SKIP_RDOC=1 SKIP_SQLDOC=1 jekyll serve --watch
```

and

```bash
cd python/docs
make clean html
```

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
I am going to proof-read and probably put some more flesh on it but it would be great if I can have some feedback about it cc @srowen, @BryanCutler, @viirya, @huaxingao, @zhengruifeng, @holdenk and @ueshin.

cc @zero323 and @Fokko as well. We'll probably have to document the guidelines about type hint stuff here.

cc @rohitmishr1484 and @fhoering who're working on different docs in PySpark.

**Review by srowen — COMMENTED** [review]
I think this is fine. It could be in the main contributing docs too but I get that this is just a bit of Python-specific commentary

**Review by zero323 — APPROVED** [review]

File: python/docs/source/development/contributing.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+Contributing by Testing Releases
+--------------------------------
+
+Before the official release, PySpark release candidates are shared in the `dev@spark.apache.org` mailing list to vote on.
+This release candidates can be easily installed via `pip`. For example, in case of Spark 3.0.0 RC1, you can install as below:
+
+.. code-block:: bash
+
+    pip install https://dist.apach e.org/repos/dist/dev/spark/v3.0.0-rc1-bin/pyspark-3.0.0.tar.gz
```
Comment (viirya) [inline python/docs/source/development/contributing.rst:36]: apach e -> apache

**Comment by huaxingao** [comment]
Looks good to me.

**Review by viirya — APPROVED** [review]
One typo found, otherwise looks good.

**Comment by holdenk** [comment]
Maybe mention in testing releases to use a throw away virtual environment?

**Comment by HyukjinKwon** [comment]
It is planned to document development environment setup separately with Conda separately so it should be fine. I could try to add some more words about that while I am here ... let me try to take a fresh look.

**Review by zhengruifeng — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
> Maybe mention in testing releases to use a throw away virtual environment?

Okay, I think it's better to add a link once the env setup page is complete. I will do it with that page.

**Comment by HyukjinKwon** [comment]
Merged to master.

Thank you guys.

**Pull request closed without merging**